### ISSUE TITLE
Hierachy differences option

### ISSUE THREAD

**Issue by mads-thines** [avillafiorita/jekyll-datapage_gen#16]
Hi!
I'm importing some data from Contentful CMS which all ends down in a yaml file with a little bit different structure than the one in your example.
I've been trying to figure out how I can get your script to work if my yaml hierachy is different.  

```
Your YAML file

- full_name: adolfo villafiorita
  first_name: adolfo
  bio: long bio goes here
- full_name: pietro molini
  first_name: pietro
  bio: another long bio
- full_name: aaron ciaghi
  first_name: aaron
  bio: another very long bio
```

```
My YAML file

page:
- sys:
    id: 2ss8tkwNri0sCmIO22EIeK
  title: Foo
  body: "Some content in a body field"
- sys:
    id: 2ss8tkwNrisdferasdfO22EIeK
  title: Bar
  body: "Some content in a body field"
```

Can I somehow change the config file to match the structure of the yaml file?

```
Your config file

page_gen-dirs: true # uncomment, if you prefer to generate named folders
page_gen:
  - data: 'member'
    template: 'profile'
    name: 'full_name'
    dir: 'people'
```

```
My config file

page_gen-dirs: true # uncomment, if you prefer to generate named folders
page_gen:
  - data: 'contentful.spaces.nets'
    template: 'accordion'
    name: 'page'
    dir: 'pages'
```

Maybe something like  "page.sys.id"
In the data you can set it to descent into the folders.
Is there a feature like that with the yaml file?

If I paste your yaml content into my yaml it works and generates the pages perfectly, but with the generated yaml file and your script it fails with this error message:
"jekyll 3.2.1 | Error:  no implicit conversion of String into Integer"

Thanks you in advance!

**Comment by flavio-elder** [comment]
hey mate did you get any luck with this!? I am in the exact same situation.

**Comment by flavio-elder** [comment]
by the way @mads-thines the solution:

`page_gen-dirs: true # uncomment, if you prefer to generate named folders
page_gen:
  - data: 'contentful.spaces.nets.page'
    template: 'accordion'
    name: 'id'
    dir: 'pages'`

**Comment by mads-thines** [comment]
@flavio-elder Thank you for your answer!
Did this also work for you when there's multiple content types?

- Thanks

**Comment by jcom09** [comment]
@flavio-elder could you please explain the answer in the context of the YAML file? 

`My YAML file

page:
- sys:
    id: 2ss8tkwNri0sCmIO22EIeK
  title: Foo
  body: "Some content in a body field"
- sys:
    id: 2ss8tkwNrisdferasdfO22EIeK
  title: Bar
  body: "Some content in a body field"`

`My config file

page_gen-dirs: true # uncomment, if you prefer to generate named folders
page_gen:
  - data: 'page.sys.id'
    template: 'accordion'
    name: 'page'
    dir: 'pages'`

page.sys.id is giving me the error: Error: no implicit conversion of String into Integer

Or @mads-thines - Were you able to get the solution working?

**Comment by filipesaraiva** [comment]
Hi, please anyone has a solution for it? My yaml file is below. I tried several combinations in `data` attribute for `page_gen` in `_config.yml` but I could not put it to work correctly. The file name for the YAML file below is `people.yml`.

```
people:
    - display_name: "Professor X"
      webpage: "http://en.wikipedia.org/wiki/Professor_X"
      github: profx
      role: faculty
      bio: |
        Professor X is the leader and founder of X-Men. 
    - display_name: "Scott Summers"
      webpage: "http://en.wikipedia.org/wiki/Cyclops_(comics)"
      role: staff
      bio: |
        Scott, also known as Cyclops, shoots lasers from his eyes, which makes him the perfect administrative assistant.
```

**Comment by avillafiorita** [comment]
Hi, sorry for taking so long to get back.

It can be done out of the box (I had to look a bit myself, before finding a solution).

Specify the full path of your data in the `data` field of `_config.yml`. That is:

     - data: 'hierarchy.people'
        template: 'profile'
        name: 'display_name'
        dir: 'hierarchy'

I have pushed an example of hierarchical data in the example site (Notice that in the example I have slightly changed your data to reuse the `profile.html` layout).

Let me know if it works, so that I can close the issue.
